lib/srv/usermgmt.go

```diff
@@ -247,9 +251,39 @@ func (u *HostUserManagement) updateUser(name string, ui services.HostUsersInfo)
 		currentGroups[group.Name] = struct{}{}
 	}
 
-	_, hasSystemGroup := currentGroups[types.TeleportServiceGroup]
-	if hasSystemGroup && ui.Mode == types.CreateHostUserMode_HOST_USER_MODE_INSECURE_DROP {
-		ui.Groups = append(ui.Groups, types.TeleportServiceGroup)
+	// allow for explicit assignment of teleport-keep group in order to facilitate migrating KEEP users that existed before we added
+	// the teleport-keep group
+	migrateKeepUser := slices.Contains(ui.Groups, types.TeleportKeepGroup)
+
+	if migrateKeepUser && ui.Mode == types.CreateHostUserMode_HOST_USER_MODE_INSECURE_DROP {
+		return trace.Errorf("can not explicitly assign %q group in 'INSECURE_DROP' mode: %w", types.TeleportKeepGroup, unmanagedUserErr)
```

Comment:
It should just ignore the rest of the request since this error embeds `unmanagedUserErr`. The caller in `sess.go` will allow the session to be created if that error is present, so we can bail out here and prevent any modifications to the unmanaged user